6.25 release notes (#3878)
astronomer/docs: astro/release-notes.md

<!-- ALL LINKS TO INTERNAL DOCS MUST BE COMPLETE URLS INCLUDING HTTPS. Otherwise the links will break in RSS. -->

## June 25, 2024

> ### Custom workload identity for AWS

Author: ```suggestion
### Customer managed workload identity for AWS
```

<HostedBadge/>

:::publicpreview
:::

The **Customer Managed Identity** Deployment setting is now available on AWS. This means that you can now assign an existing workload identity and AWS IAM role to your Airflow Deployments on Astro. When you use this setting, your Deployment uses the identity to assume the permissions of your IAM role and gain secure access to your data services. With this feature, you can:

    - Re-use or share a customer managed identity across many Deployments, either ephemeral or static.
    - Leverage existing identities when migrating from MWAA or open source Airflow environments

> This can reduce friction when migrating to Astro. See [Attach an IAM role to your Deployment](/docs/astro/authorize-deployments-to-your-cloud#attach-an-iam-role-to-your-deployment) for detailed information.

Author: ```suggestion
This can reduce friction when migrating to Astro. See [Attach an IAM role to your Deployment](https://www.astronomer.io/docs/astro/authorize-deployments-to-your-cloud#attach-an-iam-role-to-your-deployment) for detailed information.
```

### Catatonic Celery worker healer automatically addresses stuck queued tasks

<HostedBadge/>

> A new improvement to the Astro Data Plane now automatically identifies when Celery workers are onlin and healthy, but not actually processing new tasks. When this happens, it looks like many tasks are stuck in a `queued` state. With this new feature, you will experience a lower frequency of tasks stuck in a `queued` state, which cause performance and reliability issues for your Airflow implementation.

Reviewer: ```suggestion
A new improvement to the Astro Data Plane now automatically identifies when Celery workers are online and healthy, but not actually processing new tasks. When this happens, it looks like many tasks are stuck in a `queued` state. With this new feature, you will experience a lower frequency of tasks stuck in a `queued` state, which causes performance and reliability issues for your Airflow implementation.
```

The Catatonic Celery worker healer uses the following process:

> - It first identifies workers that have both tasks stuck in a `queued` state for an extensive time period and are in Deployments where the concurrency available means that tasks should not be queued.

Reviewer: ```suggestion
- It first identifies workers that both have tasks stuck in a `queued` state for an extensive time period and are in Deployments where the concurrency available means that tasks should not be queued.
```

> - The healer then kills catatonic workers that are not running tasks, or enters workers that are still running tasks into a warm shutdown period.

Reviewer: ```suggestion
- The healer then kills catatonic workers that are not running tasks or shifts workers that are still running tasks into a warm shutdown period.
```
